src/main/lrsql/util/oidc.clj

```diff
@@ -88,3 +112,67 @@
                  (throw ex)))))
          ;; no valid scopes, can't do anything
          :com.yetanalytics.lrs.auth/unauthorized)})))
+
+;; Only auth scopes
+(s/def ::scopes
+  (s/every ::scope :kind set? :into #{}))
+
+(s/def ::oidc-admin-identity
+  (s/keys :req-un [::scopes
+                   ::admin/username
+                   :lrsql.spec.admin.input/oidc-issuer]))
+
+(s/def :lrsql.util.oidc.token-auth-admin-identity/result
+  (s/nonconforming
+   (s/or :unauthorized #{::unauthorized}
+         :auth-identity
+         ::oidc-admin-identity)))
+
+(s/fdef token-auth-admin-identity
+  :args (s/cat :ctx map?
+               :scope-prefix ::config/oidc-scope-prefix)
+  :ret (s/nilable
+        :lrsql.util.oidc.token-auth-admin-identity/result))
+
+(defn token-auth-admin-identity
+  "For the given context, return a valid OIDC admin auth identity from token
+  claims.
+  args:
+    ctx - Pedestal context that may contain claims.
+    scope-prefix - Prefix to add to expected scopes."
+  [ctx
+   scope-prefix]
+  (when-let [token (:com.yetanalytics.pedestal-oidc/token ctx)]
+    (let [{:keys [scope iss sub]
+           :as   claims} (get-in ctx
+                                 [:request
+                                  :com.yetanalytics.pedestal-oidc/claims])]
+      (if-let [scopes (and (not-empty iss)
+                           (not-empty sub)
+                           (some-> scope
+                                   (parse-scope-claim
+                                    :scope-prefix scope-prefix
+                                    :scope-map oidc-scope-str-kw-map)
+                                   not-empty
+                                   (->> (into #{}))))]
+        {:scopes      scopes
+         :username    sub
+         :oidc-issuer iss}
+        ;; no valid scopes, can't do anything
+        ::unauthorized))))
+
+(s/fdef authorize-admin-action
+  :args (s/cat :ctx           (s/keys :req-un [::auth/request])
+               :auth-identity (s/keys :req-un [::scopes]))
+  :ret (s/keys :req-un [::auth/result]))
+
+(defn authorize-admin-action
+  "Given a pedestal context and an OIDC admin auth identity, authorize or deny."
+  [{{:keys [request-method path-info]} :request
```

Comment:
`request-method` is an unused binding.

Reply:
https://github.com/yetanalytics/lrsql/pull/207/commits/d2db9b892d0a5df1e39c6d2a7609f76d52f79493